backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/bpartner/bpartnercomposite/JsonRetrieverService.java

```diff
@@ -748,6 +774,46 @@ Optional<BPartnerComposite> getBPartnerCompositeAssertCacheHit(@NonNull final Im
 		return extractResult(bpartnerComposites);
 	}
 
+	@NonNull
+	public Optional<MetasfreshId> resolveBankAccountId(
+			@NonNull final ExternalIdentifier bankAccountIdentifier,
+			@NonNull final BPartnerComposite bPartnerComposite)
+	{
+		Check.assumeNotNull(bPartnerComposite.getOrgId(), "At this point, the orgId is always resolved on BPartnerComposite!");
+
+		return switch (bankAccountIdentifier.getType())
+				{
+					case METASFRESH_ID -> Optional.of(bankAccountIdentifier.asMetasfreshId());
+					case EXTERNAL_REFERENCE -> externalReferenceService.getJsonMetasfreshIdFromExternalReference(bPartnerComposite.getOrgId(),
```

Comment:
i suggest to introduce BPartnerComposite.getOrgIdNotNull() and do the not null validation there